In [1]:
from google.colab import files
uploaded = files.upload()

Saving car_dataset.csv to car_dataset.csv


In [3]:
import pandas as pd
df = pd.read_csv("car_dataset.csv")

In [4]:
import pandas as pd

df = pd.read_csv("car_dataset.csv")  # or your file name

df.head()
df.info()
df.describe()
df.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8128 entries, 0 to 8127
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   name           8128 non-null   object 
 1   year           8128 non-null   int64  
 2   selling_price  8128 non-null   int64  
 3   km_driven      8128 non-null   int64  
 4   fuel           8128 non-null   object 
 5   seller_type    8128 non-null   object 
 6   transmission   8128 non-null   object 
 7   owner          8128 non-null   object 
 8   mileage        7907 non-null   object 
 9   engine         7907 non-null   object 
 10  max_power      7913 non-null   object 
 11  torque         7906 non-null   object 
 12  seats          7907 non-null   float64
dtypes: float64(1), int64(3), object(9)
memory usage: 825.6+ KB


(8128, 13)

In [8]:
df["brand"] = df["name"].str.split().str[0]
df["brand"] = df["brand"].str.strip().str.lower()

In [9]:
df = df.dropna(subset=["selling_price"])

In [10]:
df["mileage"] = df["mileage"].astype(str).str.extract(r'(\d+\.?\d*)')
df["mileage"] = pd.to_numeric(df["mileage"], errors="coerce")

In [11]:
df["engine"] = df["engine"].astype(str).str.extract(r'(\d+\.?\d*)')
df["engine"] = pd.to_numeric(df["engine"], errors="coerce")

df["max_power"] = df["max_power"].astype(str).str.extract(r'(\d+\.?\d*)')
df["max_power"] = pd.to_numeric(df["max_power"], errors="coerce")

df["torque"] = df["torque"].astype(str).str.extract(r'(\d+\.?\d*)')
df["torque"] = pd.to_numeric(df["torque"], errors="coerce")

In [12]:
numeric_cols = ["mileage", "engine", "max_power", "torque", "seats"]

for col in numeric_cols:
    df[col] = df[col].fillna(df[col].mean())

In [13]:
categorical_cols = ["fuel", "seller_type", "transmission", "owner", "brand"]

for col in categorical_cols:
    df[col] = df[col].fillna(df[col].mode()[0])

In [15]:
df = df.drop_duplicates()
print("Duplicates:", df.duplicated().sum())
df = df.drop_duplicates()
print("Shape after removing duplicates:", df.shape)

Duplicates: 0
Shape after removing duplicates: (6926, 14)


In [16]:
df.info()
df.describe()
df.shape

<class 'pandas.core.frame.DataFrame'>
Index: 6926 entries, 0 to 8125
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   name           6926 non-null   object 
 1   year           6926 non-null   int64  
 2   selling_price  6926 non-null   int64  
 3   km_driven      6926 non-null   int64  
 4   fuel           6926 non-null   object 
 5   seller_type    6926 non-null   object 
 6   transmission   6926 non-null   object 
 7   owner          6926 non-null   object 
 8   mileage        6926 non-null   float64
 9   engine         6926 non-null   float64
 10  max_power      6926 non-null   float64
 11  torque         6926 non-null   float64
 12  seats          6926 non-null   float64
 13  brand          6926 non-null   object 
dtypes: float64(5), int64(3), object(6)
memory usage: 811.6+ KB


(6926, 14)

In [17]:
from sklearn.metrics import mean_absolute_error

y_true = df["selling_price"]
baseline_prediction = y_true.mean()
y_pred = [baseline_prediction] * len(y_true)

mae = mean_absolute_error(y_true, y_pred)

print("Baseline prediction (mean selling_price):", baseline_prediction)
print("Baseline MAE:", mae)

Baseline prediction (mean selling_price): 517270.6784579844
Baseline MAE: 292164.1402292158


1. Exploration findings
*   The dataset contains 8128 rows and 13 columns.
*   Several columns such as mileage, engine, max_power, and torque were stored as object instead of numeric.
*   Missing values were found in mileage, engine, max_power, torque, and seats.
*   A brand column was derived from the name column and standardized.
*   Duplicate rows were removed.

2. Cleaning actions
*   Dropped rows with null target values in selling_price
*   Extracted numeric values from mileage, engine, max_power, and torque
*   Imputed missing feature values
*   Standardized brand text using strip and lowercase
*   Removed duplicates

3. Baseline model
*   Built a baseline model that predicts the mean selling price for every record
*   Calculated MAE to use as a benchmark before applying machine learning models



